rust/src/dns/detect.rs

```diff
@@ -53,6 +56,23 @@ fn parse_opcode(opcode: &str) -> Result<DetectDnsOpcode, ()> {
     Err(())
 }
 
+fn parse_rcode(rcode: &str) -> Result<DetectDnsRcode, ()> {
```

Comment:
Reusing `DetectUintData<u16>` supplies you the parsing function `rs_detect_u16_parse`

Reply:
(and this allows more features, like negation, lesser or greater than, etc...)

Reply:
I was wondering if rcode can be negated? Should I add functions and tests to check that as well similar to opcode?

Reply:
I think it makes sense to have negation on the `rcode`.  I might want to alert on all non-zero rcodes without listing them all for example.